modulus/datapipes/gnn/drivaernet_dataset.py

```diff
@@ -0,0 +1,391 @@
+                ),
+            }.items()
+        }
+
+        self.estats = {
+            "x": {
+                "mean": torch.tensor([0, 0, 0, 0.01338306]),
+                "std": torch.tensor([0.00512953, 0.00953013, 0.00923065, 0.00482016]),
+            }
+        }
+
+    def __len__(self) -> int:
+        return len(self.coeffs)
+
+    def __getitem__(self, idx: int) -> dgl.DGLGraph:
+        if not 0 <= idx < len(self):
+            raise IndexError(f"Invalid {idx = }, must be in [0, {len(self)})")
+
+        coeffs = self.coeffs.iloc[idx]
+        gname = coeffs.name
+
+        if self.cache_dir is None:
+            # Caching is disabled - create the graph.
+            graph = self._create_dgl_graph(gname)
+        else:
+            cached_graph_filename = self.cache_dir / (gname + ".bin")
+            if not self._force_reload and cached_graph_filename.is_file():
+                gs, _ = dgl.load_graphs(str(cached_graph_filename))
+                if len(gs) != 1:
+                    raise ValueError(f"Expected to load 1 graph but got {len(gs)}.")
+                graph = gs[0]
+            else:
+                graph = self._create_dgl_graph(gname)
+                dgl.save_graphs(str(cached_graph_filename), [graph])
+
+        # Set graph inputs/outputs.
+        graph.ndata["x"] = torch.cat([graph.ndata[k] for k in self.input_keys], dim=-1)
+        graph.ndata["y"] = torch.cat([graph.ndata[k] for k in self.output_keys], dim=-1)
+
+        return {
+            "name": gname,
+            "graph": graph,
+            "c_d": torch.tensor(coeffs["Average Cd"], dtype=torch.float32),
+        }
+
+    @staticmethod
+    def _get_cache_dir(data_dir, cache_dir):
+        if not cache_dir.is_absolute():
+            cache_dir = data_dir / cache_dir
+        return cache_dir.resolve()
+
+    def _create_dgl_graph(
+        self,
+        name: str,
+        to_bidirected: bool = True,
+        dtype: torch.dtype | str = torch.int32,
+    ) -> dgl.DGLGraph:
+        """Creates a DGL graph from DrivAerNet VTK data.
+
+        Parameters
+        ----------
+        name : str
+            Name of the graph in DrivAerNet.
+        to_bidirected : bool, optional
+            Whether to make the graph bidirected. Default is True.
+        dtype : torch.dtype or str, optional
+            Data type for the graph. Default is torch.int32.
+
+        Returns
+        -------
+        dgl.DGLGraph
+            The DGL graph.
+        """
+
+        def extract_edges(mesh: pv.PolyData) -> list[tuple[int, int]]:
+            # Extract connectivity information from the mesh.
+            # Traversal API is faster comparing to iterating over mesh.cell.
+            polys = mesh.GetPolys()
+            if polys is None:
+                raise ValueError("Failed to get polygons from the mesh.")
+
+            polys.InitTraversal()
+
+            edge_list = []
+            for _ in range(polys.GetNumberOfCells()):
```

Comment:
I think this piece of code does not capture all the edges. (Can we verify the edges generated from this matches with what VTK graph has natively?). For the Stokes dataset, I had to change the following to (I am yet to submit a PR for that):

```
id_list = vtk.vtkIdList()
for _ in range(polys.GetNumberOfCells()):
    polys.GetNextCell(id_list)
    num_ids = id_list.GetNumberOfIds()
    for j in range(num_ids):
        edge_list.append((id_list.GetId(j), id_list.GetId(j+1) % num_ids))
```

The `% num_ids` was required to close the loop and get all the connections / edges of a triangle. 

cc. @mnabian

Reply:
![image](https://github.com/NVIDIA/modulus/assets/71059996/ada7c9d7-5ecd-4c73-b760-8960da0a38f7)

Reply:
Probably does not matter for graph training, but can become important if we expect the graph to be used in further downstream tasks such as computing gradients for which having accurate connectivity information is important.

Reply:
Ref: https://github.com/NVIDIA/modulus/pull/549

Reply:
Also this PR proposes another fix: https://github.com/NVIDIA/modulus/pull/561

Reply:
@ktangsali @mnabian I like the proposed solution better than `% num_ids` - it makes it clearer of what's going on, so I've incorporated it into my code.

Reply:
I am happy as long as all the edges are included :)